unit_tests/testLinearSolver/testLinearSolverParallel.f90

```diff
@@ -0,0 +1,206 @@
+  REAL(SRK),POINTER :: thisX(:)
+  INTEGER(SIK) :: i
+  LOGICAL(SBK) :: match, bool
+
+  ALLOCATE(LinearSolverType_Iterative :: thisLS)
+
+  COMPONENT_TEST('DistributedBandedMatrixType')
+  !With GMRES
+  !The sparse matrix type
+
+  ! initialize linear system
+  CALL pList%clear()
+  CALL pList%add('LinearSolverType->TPLType',LS_NATIVE)
+  CALL pList%add('LinearSolverType->solverMethod',GMRES)
+  CALL pList%add('LinearSolverType->MPI_Comm_ID',PE_COMM_WORLD)
+  CALL pList%add('LinearSolverType->numberOMP',1_SNK)
+  CALL pList%add('LinearSolverType->timerName','testTimer')
+  CALL pList%add('LinearSolverType->matType',DISTRIBUTED_BANDED)
+  CALL pList%add('LinearSolverType->A->MatrixType->n',9_SNK)
+  CALL pList%add('LinearSolverType->A->MatrixType->m',9_SNK)
+  CALL pList%add('LinearSolverType->A->MatrixType->nnz',33_SNK)
+  CALL pList%add('LinearSolverType->x->VectorType->n',9_SNK)
+  CALL pList%add('LinearSolverType->b->VectorType->n',9_SNK)
+  CALL pList%validate(pList)
+  CALL thisLS%init(pList)
+
+  !A =  4    -1     0    -1     0     0     0     0     0
+  !    -1     4    -1     0    -1     0     0     0     0
+  !     0    -1     4     0     0    -1     0     0     0
+  !    -1     0     0     4    -1     0    -1     0     0
+  !     0    -1     0    -1     4    -1     0    -1     0
+  !     0     0    -1     0    -1     4     0     0    -1
+  !     0     0     0    -1     0     0     4    -1     0
+  !     0     0     0     0    -1     0    -1     4    -1
+  !     0     0     0     0     0    -1     0    -1     4
+  SELECTTYPE(A => thisLS%A); TYPE IS(DistributedBandedMatrixType)
+      CALL A%set(1,1, 4.0_SRK)
+      CALL A%set(1,2,-1.0_SRK)
+      CALL A%set(1,4,-1.0_SRK)
+      CALL A%set(2,1,-1.0_SRK)
+      CALL A%set(2,2, 4.0_SRK)
+      CALL A%set(2,3,-1.0_SRK)
+      CALL A%set(2,5,-1.0_SRK)
+      CALL A%set(3,2,-1.0_SRK)
+      CALL A%set(3,3, 4.0_SRK)
+      CALL A%set(3,6,-1.0_SRK)
+      CALL A%set(4,1,-1.0_SRK)
+      CALL A%set(4,4, 4.0_SRK)
+      CALL A%set(4,5,-1.0_SRK)
+      CALL A%set(4,7,-1.0_SRK)
+      CALL A%set(5,2,-1.0_SRK)
+      CALL A%set(5,4,-1.0_SRK)
+      CALL A%set(5,5, 4.0_SRK)
+      CALL A%set(5,6,-1.0_SRK)
+      CALL A%set(5,8,-1.0_SRK)
+      CALL A%set(6,3,-1.0_SRK)
+      CALL A%set(6,5,-1.0_SRK)
+      CALL A%set(6,6, 4.0_SRK)
+      CALL A%set(6,9,-1.0_SRK)
+      CALL A%set(7,4,-1.0_SRK)
+      CALL A%set(7,7, 4.0_SRK)
+      CALL A%set(7,8,-1.0_SRK)
+      CALL A%set(8,5,-1.0_SRK)
+      CALL A%set(8,7,-1.0_SRK)
+      CALL A%set(8,8, 4.0_SRK)
+      CALL A%set(8,9,-1.0_SRK)
+      CALL A%set(9,6,-1.0_SRK)
+      CALL A%set(9,8,-1.0_SRK)
+      CALL A%set(9,9, 4.0_SRK)
+      CALL A%assemble()
+  ENDSELECT
+
+  ! Use default initial guess (all ones)
+  SELECTTYPE(x => thisLS%x); TYPE IS(NativeDistributedVectorType)
+    x%b = 1.0_SRK
+  ENDSELECT
+
+  SELECTTYPE(b => thisLS%b); TYPE IS(NativeDistributedVectorType)
+    b%b = 1.0_SRK
+  ENDSELECT
```

Comment:
Assertions for if the LS is initialized and A is setup properly?

Reply:
Since the parallel solver is the same object as the serial one (just with parallel datatypes) I figured that duplicating the initialization tests (and setup for parallel datatypes) would be overdoing it.
I can add them, but anything wrong with the initialization should be caught in the serial test.